R/db.R

```diff
@@ -284,11 +284,11 @@ write_data_safely <- function(dbconn, tblname, df, key, safe_columns, blacklist)
     if (length(blacklist) > 0L) {
       df_without_id <- df[-which(names(df) == key)]  # Don't blacklist on id.
       if (any(is.na(blacklist))) {
-        df <- df[apply(!is.na(df_without_id), 1, all),]
+        df <- df[apply(!is.na(df_without_id), 1, all), , drop = FALSE]
         blacklist <- blacklist[!is.na(blacklist)]
       }
       if (length(blacklist) > 0L) {
-        df <- df[apply(df_without_id, 1, function(x) all(!(x %in% blacklist))),]
+        df <- df[apply(df_without_id, 1, function(x) all(!(x %in% blacklist))), , drop = FALSE]
```

Comment:
Sounds like a good idea. We also need a more efficient df -> df function than `apply`. But I think we can do either thing in a future PR.